src/OrleansProviders/Streams/Common/PooledCache/IEvictionStrategy.cs

```diff
@@ -28,6 +28,15 @@ public interface IEvictionStrategy
         /// </summary>
         /// <param name="newBlock"></param>
         void OnBlockAllocated(FixedSizeBuffer newBlock);
+
+        /// <summary>
+        /// Given a purge cached message, indicates whether it should be purged from the cache.
+        /// </summary>
+        /// <param name="cachedMessage"></param>
+        /// <param name="newestCachedMessage"></param>
+        /// <param name="nowUtc"></param>
+        /// <returns></returns>
+        bool ShouldPurge(ref CachedMessage cachedMessage, ref CachedMessage newestCachedMessage, DateTime nowUtc);
```

Comment:
Why add this to the abstraction.  Is called by other?  Shouldn't currently be, as it was private.